fcrepo-kernel-modeshape/src/main/java/org/fcrepo/kernel/modeshape/FedoraResourceImpl.java

```diff
@@ -432,6 +407,50 @@ public void delete() {
         }
     }
 
+    private void removeReferences(final Node n) {
+        try {
+            // Remove references to this resource
+            doRemoveReferences(n);
+
+            // Recurse over children of this resource
+            if (n.hasNodes()) {
+                iteratorToStream(n.getNodes()).forEach(child -> removeReferences((Node) child));
```

Comment:
That makes sense, @acoburn. Is that a better approach than what we currently have in this PR?